src/batch/src/executor/row_seq_scan.rs

```diff
@@ -138,25 +139,52 @@ impl BoxedExecutorBuilder for RowSeqScanExecutorBuilder {
             NodeBody::RowSeqScan
         )?;
 
+        let table_desc: &CellBasedTableDesc = seq_scan_node.get_table_desc()?;
         let table_id = TableId {
-            table_id: seq_scan_node.table_desc.as_ref().unwrap().table_id,
+            table_id: table_desc.table_id,
         };
-        let column_descs = seq_scan_node
-            .column_descs
+        let column_descs = table_desc
+            .columns
             .iter()
-            .map(|column_desc| ColumnDesc::from(column_desc.clone()))
+            .map(ColumnDesc::from)
             .collect_vec();
-        let pk_descs_proto = &seq_scan_node.table_desc.as_ref().unwrap().order_key;
-        let pk_descs: Vec<OrderedColumnDesc> = pk_descs_proto.iter().map(|d| d.into()).collect();
+        let column_ids = seq_scan_node
+            .column_ids
+            .iter()
+            .copied()
+            .map(ColumnId::from)
+            .collect();
+
+        let pk_descs: Vec<OrderedColumnDesc> =
+            table_desc.order_key.iter().map(|d| d.into()).collect();
         let order_types: Vec<OrderType> = pk_descs.iter().map(|desc| desc.order).collect();
 
+        let pk_indices = pk_descs
```

Comment:
Can we pass pk_indices from frontend? just like top_n or join.

Reply:
You are very forward-looking! Plan to do this in next PRs. cc @st1page

Reply:
🥵🥵 foward 🥵🥵 looking!